st2client/st2client/commands/keyvalue.py

```diff
@@ -77,23 +80,40 @@ def __init__(self, *args, **kwargs):
                                  help='Scope item is under. Example: "user".')
         self.parser.add_argument('-u', '--user', dest='user', default=None,
                                  help='User for user scoped items (admin only).')
+        self.parser.add_argument('-n', '--last', type=int, dest='last',
+                                 default=self.default_limit,
+                                 help=('List N most recent %s. Default is 50.' %
+                                       self.resource_name))
 
-    def run_and_print(self, args, **kwargs):
+    @resource.add_auth_token_to_kwargs_from_cli
+    def run(self, args, **kwargs):
+        # Filtering options
         if args.prefix:
             kwargs['prefix'] = args.prefix
-
         decrypt = getattr(args, 'decrypt', False)
-        kwargs['params'] = {'decrypt': str(decrypt).lower()}
```

Comment:
This is what happens when params is used:
```
>>> kwargs={'token': u'c992ce735afd496fb9c17fa0499ce3a2', 'params': {'scope': 'system', 'decrypt': 'false', 'user': None}}
>>> token = kwargs.get('token', None)
  2 api_key = kwargs.get('api_key',None)
  3 params = kwargs.get('params', {})
>>> for k, v in six.iteritems(kwargs):
  2             # Note: That's a special case to support api_key and token kwargs
  3             if k not in ['token', 'api_key']:
  4                 params[k] = v
>>> params
{'scope': 'system', 'decrypt': 'false', 'user': None, 'params': {...}}
>>> urllib.parse.urlencode(params)
'scope=system&decrypt=false&user=None&params=%7B%27scope%27%3A+%27system%27%2C+%27decrypt%27%3A+%27false%27%2C+%27user%27%3A+None%2C+%27params%27%3A+%7B...%7D%7D'
>>>
```
here: https://github.com/StackStorm/st2/blob/master/st2client/st2client/models/core.py#L251